### PR TITLE
adds legendSort method

### LINKED ISSUE #116: add legendSort method
The `Viz` class should have a method called `legendSort`, which is a standard `(a, b)` JavaScript comparator function that gets applied to `legendData` here: https://github.com/d3plus/d3plus-viz/blob/master/src/_drawLegend.js#L62

### PR DESCRIPTION
**Pull request by rbaheti**
### Description

Added a legendSort method which sorts the legends based on the user key
closes #116

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have documented all new methods.
- [ ] I have written tests for all new methods/functionality.
- [ ] I have written examples for all new methods/functionality.

### CODE REVIEW

File: src/_drawLegend.js [hunk 1]
```diff
@@ -60,6 +60,8 @@ export default function(data = []) {
       .rollup(leaves => legendData.push(merge(leaves, this._aggs)))
       .entries(this._colorScale ? data.filter((d, i) => this._colorScale(d, i) === undefined) : data);
 
+    legendData.sort((a, b) => this._legendSort(a, b));
```
Comment (davelandry) [inline src/_drawLegend.js]: no need to wrap this in a function:

```js
legendData.sort(this._legendSort);
```

File: src/Viz.js [hunk 2]
```diff
@@ -1092,6 +1093,16 @@ function value(d) {
     return arguments.length ? (this._legendPosition = _, this) : this._legendPosition;
   }
 
+  /**
+      @memberof Viz
+      @desc A JavaScript [sort comparator function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) for legends.
```
Comment (davelandry) [inline src/Viz.js]:
```js
@desc A JavaScript [sort comparator function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) used to sort the legend.
```

File: src/Viz.js [hunk 3]
```diff
@@ -132,6 +132,7 @@ export default class Viz extends BaseClass {
     this._legendCutoff = 1;
     this._legendPadding = defaultPadding;
     this._legendPosition = "bottom";
+    this._legendSort = (a, b) => a - b;
```
Comment (davelandry) [inline src/Viz.js]: ...does this actually work? aren't `a` and `b` JavaScript data objects? Can you try to add a default sorting that uses the `drawLabel` function to sort alphabetically (this is just pseudo-code, I haven't tested this):

```js
this._drawLabel(a).localeCompare(this._drawLabel(b))
```

Comment (rbaheti) [reply src/Viz.js]: Right, this wasn't working by default. But I wasn't sure which data key to use to compare objects. I tested it by overriding legendSort while creating the plot.

Comment (rbaheti) [reply src/Viz.js]: `drawLabel` function takes 2 parameters, `d, i`, while we have only `d`. Should I just pass `d` then? That seems wrong though.

Comment (davelandry) [reply src/Viz.js]: I think, at least for now, it's okay to not pass `i` (as long as it works with the default settings!)

**Pull request merged**